### ISSUE TITLE
STE的实现问题

### ISSUE THREAD

**Issue by daimal** [XiaoLongtaoo/TIGER#16]
RQ-VAE 后续层 (Layer 2+) Commitment Loss 梯度未回传至 Encoder（残差计算中 STE 导致梯度断裂），在当前 RQ-VAE（Residual Quantized VAE）实现中，第 2 层及后续所有层级的 Commitment Loss 无法将梯度回传至 Encoder 输出（z_base）。
虽然测试总 Loss（如 total_loss = mean([loss_0, loss_1])）时梯度不为零，但通过单独提取第 2 层 Loss (loss_1) 对 z_base 求导可以证实，来自第 2 层的 Commitment Loss 梯度实际上恒等于 0。这是bug？还是有另外的动机？
